lib/note-content-editor.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { invoke } from 'lodash';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState )
+		}
+	},
+
+	saveEditorRef( ref ) {
+		this.editor = ref
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
+
+		this.setState( { editorState }, () => {
+			if ( contentHasChanged ) {
+				this.props.onChangeContent( plainTextContent( editorState ) );
```

Comment:
if we use the helper variable above, we can just pass it here too -> `onChangeContent( nextContent )`

also, we could rearrange this here with another variable. it's styling purely; my preference would be something like this…

```js
import { noop } from 'lodash';

const announceChanges = contentHasChanged
	? () => this.props.onChangeContent( nextContent )
	: noop;
```

in fact, if that's the only thing that we use `hasContentChanged` for, we could join the expressions…

```js
const announceChanges = nextContent !== prevContent
	? () => this.props.onChangeContent( nextContent )
	: noop;

this.setState( { editorState }, announceChanges );
```